plugins/pulp_rpm/plugins/importers/yum/purge.py

```diff
@@ -393,7 +393,9 @@ def _duplicate_key_id_generator_aggregation(unit, fields):
 
     # When aggregating over hundreds of thousands of packages, mongo can overflow
     # To prevent this, mongo needs to be allowed to temporarily use the disk for this transaction
-    aggregation = unit.objects.aggregate(sort, project, allowDiskUse=True)
+    # Set the batch size to 5 to prevent a cursor timeout
+    aggregation = unit.objects.aggregate(sort, project, allowDiskUse=True,
+                                         batchSize=5)
```

Comment:
Given the max document size of 16MB and (I think) that we know the max length of the nevra fields, I wonder if we can come up with a calculated batch size that further improves the performance here.

That said, I'm missing the connection between increasing the amount of work done in mongo (and I assume the time taken to do that work), and preventing a cursor timeout. Can you elabortate?